### ISSUE TITLE
Jenkins -> Rundeck - possible plugin issue with dev code ( works with 2.5.2 builds )

### ISSUE THREAD

**Issue by ddzed15** [rundeck/rundeck#1454]
Hello,
We started to notice that the jenkins trigger to execute Rundeck jobs started to fail with the 2.5.3 code update ( we are compiling the source based on the last released 2.5.3 Git from development branch). 
Could this be related to the recent support for JSON API in 2.5.3? 
`
(API v14) If the request has Content-Type: application/json, then the parameters will be ignored, and this format is expected in the content:
`

Works in the 2.5.2 build. 
Works in 2.5.3 build calling the API directly ( via Curl or REST Call) with the same URI that fails within jenkins. 
Jenkins plugin is configured correctly ( https://rundeck.example.net/rundeck as the URI) and the Token works and Jenkins is able to connect to the server. 

Below is the error:
`
16:47:34 [EnvInject] - Variables injected successfully.
16:47:34 Notifying Rundeck...
16:47:34 Error while talking to Rundeck's API at https://rundeck.example.net/rundeck : Invalid HTTP response 'HTTP/1.1 405 Method Not Allowed' for https://rundeck.example.net/rundeck/api/12/job/8ad979a6-6c82-4458-b236-b1e4566f5156/run?argString=-BASELINE+http%3A%2F%2Fdev.example.net%2Fjenkins%2Fjob%2Frundeck-development-clone%2F9%2F%2Fartifact%2Frundeckapp%2Fbuild%2Fdistributions%2Frundeck-2.5.3.zip+-VERSION+2.5.3-9&hostname=hostname.example.net
16:47:34 Build step 'Rundeck' marked build as failure
`
We changed the API version in the jenkins main config to 14 but still the same error.

Below is a success from the 2.5.2 version
`
17:08:17 [EnvInject] - Variables injected successfully.
17:08:18 Notifying Rundeck...
17:08:18 Notification succeeded ! Execution #2479, at https://rundeck.example.net/rundeck/execution/follow/2479 (status : RUNNING)
17:08:18 Waiting for Rundeck execution to finish...
17:08:24 Rundeck execution #2479 finished in 3 seconds, with status : SUCCEEDED
`
Cheers, Dharmendra

**Comment by ddzed15** [comment]
Hello,
We found an issue and that seems to have fixed it.
The GET method on apiJobRun was removed as part of the  commit below. 
Adding the GET back to the allowedMethods and testing,  it seems to work fine. 
```javascript
101c101
<             apiJobRun: ['POST','GET'],
---
>             apiJobRun: ['POST'],
```
https://github.com/rundeck/rundeck/commit/c17664702ac2e90106d7f6fc3a26371eb2e540aa#diff-6eae952191d719a4612369e6dbca725c

Not sure if the real fix should be in the Jenkins Rundeck plugin, but this works for us now. 

Cheers, Dharmendra

**Comment by gschueler** [comment]
ok, i think that was a regression when we moved to API v14.  GET should be supported up to v13.  The jenkins plugin uses the rundeck-api-java-client lib, which is not yet updated for api V14, so that needs to be updated as well.

**Comment by gschueler** [comment]
Sorry, I think you are using the Development branch (unreleased), not 2.5.3 release version.

I will update the milestone to 2.6.0.

Thanks

**Comment by gschueler** [comment]
fix in a53dead1f3c2b9833fcc7844b92d65de0ff6555d
